low-level/framework/utils/iem.py

```diff
@@ -155,6 +153,20 @@ def check_exsisting_fault_iems(self):
             if prev_fault_iem_event:
                 self.fault_iems.append(event_name)
 
-def generate_iem(iem):
-    """Generate iem."""
-    syslog.syslog(iem)
+    @staticmethod
+    def generate_iem(module, event_code, severity, description):
+        """Generate iem and send it to a MessgaeBroker."""
+        try:
+            logger.info(f"Sending IEM alert for module:{module}"
+                        f" and event_code:{event_code}")
+            # check if IEM Framework initialized,
+            # if not, retry initializing the IEM Frameowork
+            if os.path.exists(IEM_INIT_FILE):
+                EventMessage.init(component='sspl', source='S')
```

Comment:
what if init fails here too ? Then log it and return after checking for init failure marker file itself dont proceed for send()
One problem though, see if we can avoid that - this file existence check will happen before every IEM generation, so its a overhead even if Init had passed at first time itself.

Reply:
this code is already is try except block.
if init fails here as well, it will go to exception block and log an error.
send api will not be called in that case.

Reply:
rather than checking init_failed file before every send api, should we move this code of class's __init__ method?
in this way, same obj if sends different iem alerts, only once file exists or not will be checked.
@cdeshmukh should I do this way or any better approach suggestions?